core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -238,6 +245,9 @@ class KafkaApis(val requestChannel: RequestChannel,
     // stop serving data to clients for the topic being deleted
     val stopReplicaRequest = request.body[StopReplicaRequest]
     apisUtils.authorizeClusterOperation(request, CLUSTER_ACTION)
+    if (controller == null) {
```

Comment:
Can we rename "controller" to "legacyController" to make it more clear what's going on here?  We can do that in a separate PR if it's easier